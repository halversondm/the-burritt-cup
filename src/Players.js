import React, { Component } from 'react';

class Players extends Component {

    render() {
        return (
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <h1>Players</h1>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/owen_full.jpg">
                                        <img className="head" src="./img/thumbnails/owen_head.jpg" alt="Owen Burritt" />
                                    </a> Owen Burritt
                    </div>
                            </h2>
                            <h3 className="post-subtitle">
                                Burritt Cup Commissioner
                </h3>
                            <h4 className="post-subtitle">
                                Hometown: Bartlett, Illinois <br /> Nickname: Hook<br />

                            </h4>
                            <p className="post-meta">Owen likes to hit left. Hence the nickname.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/kyle_full.jpg">
                                        <img className="head" src="./img/thumbnails/kyle_head.jpg" alt="Kyle Burritt" />
                                    </a> Kyle Burritt
                    </div>
                            </h2>
                            <h3 className="post-subtitle">
                                Burritt Cup Rules Committee
                </h3>
                            <h4 className="post-subtitle">
                                Hometown: Lombard, Illinois <br /> Nickname: Hooteria
                </h4>
                            <p className="post-meta">As the story goes, Kyle earned his nickname for a run in with some bad Hooters
                                wings. He had to make
                    an emergency stop during the first hole during the 2014 Cup Weekend.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/clint_full.jpg">
                                        <img className="head" src="./img/thumbnails/clint_head.jpg" alt="Clint Delee" />
                                    </a> Clint Delee
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Bartlett, Illinois <br /> Nickname: Loud
                </h4>
                            <p className="post-meta">HELLO! Clint is famous for making loud responses or comments so the nickname is
                    well earned.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/nat_full.jpg">
                                        <img className="head" src="./img/thumbnails/nat_head.jpg" alt="Nat DeRose" />
                                    </a> Nat DeRose
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Wheaton, Illinois <br /> Nickname: Up All Night
                </h4>
                            <p className="post-meta">Nat seems to be the first one to fall asleep during our activities. During the 2014
                                Cup Weekend, he fell
                    asleep outside at a fire pit behind the hotel.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/kevin_full.jpg">
                                        <img className="head" src="./img/thumbnails/kevin_head.jpg" alt="Kevin Eck" />
                                    </a> Kevin Eck
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Berwyn, Illinois <br /> Nickname: Pants
                </h4>
                            <p className="post-meta">Kevin is quite possibly the best dressed player of the group. I believe he
                                mentioned that he's got 14
                    pairs of golf shoes. SHOES! You'll never see him in shorts so he earned the name Pants easily.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/dan_full.jpg">
                                        <img className="head" src="./img/thumbnails/dan_head.jpg" alt="Dan Halverson" />
                                    </a> Dan Halverson
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Aurora, Illinois <br /> Nickname: Shotgun
                </h4>
                            <p className="post-meta">Click-Click-Pull ... or what Dan's flat stick sounds like on the greens hence the nickname.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/jason_full.jpg">
                                        <img className="head" src="./img/thumbnails/jason_head.jpg" alt="Jason Lane" />
                                    </a> Jason Lane
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Plainfield, Illinois <br /> Nickname: Bear
                </h4>
                            <p className="post-meta">He snores like a Bear so the nickname. Current reports during the 2016 Cup Weekend
                                point to an alleviation
                    of the snoring due to 'My Pillow'. He may need a new nickname.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/dave_full.jpg">
                                        <img className="head" src="./img/thumbnails/dave_head.jpg" alt="Dave Kakareka" />
                                    </a> Dave Kakareka
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Chicago, Illinois <br /> Nickname: Firepit
                </h4>
                            <p className="post-meta">Finally a nickname!</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/jacob_full.jpg">
                                        <img className="head" src="./img/thumbnails/jacob_head.jpg" alt="Jacob Riebe" />
                                    </a> Jacob Riebe
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: New Lenox, Illinois <br /> Nickname: O-ke-doke
                </h4>
                            <p className="post-meta">You know, like the cheese popcorn. Jake really likes it and ate a bag in one
                    sitting during a Cup Weekend.</p>
                        </div>
                    </div>
                </div>
                <hr />
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/sean_full.jpg">
                                        <img className="head" src="./img/thumbnails/sean_head.jpg" alt="Sean Smith" />
                                    </a> Sean Smith
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Bartlett, Illinois <br /> Nickname: Irish Nachos
                </h4>
                            <p className="post-meta">Sean made his Cup debut in 2017. He even took home an award in his first year out.
                                We'll be looking for
                                lower scores next year and more hardware for Sean to take home. No nickname yet as he flew under
                    the radar of the Crew's watchful eye for strange behavior.</p>
                        </div>
                    </div>
                </div>
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/mike_full.jpg">
                                        <img className="head" src="./img/thumbnails/mike_head.jpg" alt="Mike Robey" />
                                    </a> Mike Robey
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Casey, Illinois <br /> Nickname: Libman
                </h4>
                            <p className="post-meta">Mike made his Cup debut in 2018. In his first year, he took home the prestigious
                    broken club, High Round Award.</p>
                        </div>
                    </div>
                </div>
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/jim_full.jpg">
                                        <img className="head" src="./img/thumbnails/jim_head.jpg" alt="Jim Kopitke" />
                                    </a> Jim Kopitke
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Bartlett, Illinois <br /> Nickname: Red Eye
                </h4>
                            <p className="post-meta">Jim made his Cup debut in 2018. Jim's nickname comes from the fact that 'red eyed'
                    back to Chicago from LAX to be part of the Burritt Cup. We like dedication!</p>
                        </div>
                    </div>
                </div>
                <div className="row justify-content-center">
                    <div className="col-lg-8 col-md-10">
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href="img/full/justin_full.jpg">
                                        <img className="head" src="./img/thumbnails/justin_head.jpg" alt="Justin Niekamp" />
                                    </a> Justin Niekamp
                    </div>
                            </h2>
                            <h4 className="post-subtitle">
                                Hometown: Pingree Grove, Illinois <br /> Nickname: TBD
                </h4>
                            <p className="post-meta">Justin made his Cup debut in 2018.</p>
                        </div>
                    </div>
                </div>
            </div>
        );
    }

}

export default Players;