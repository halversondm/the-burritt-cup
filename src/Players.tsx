import {useState, useEffect} from 'react';
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import Player from './Player.js';

export default function Players() {

    const [state, setState] = useState({
        players: [
            {
                image: "",
                thumbnail: "",
                alt: "",
                name: "",
                office: "",
                hometown: "",
                nickname: "",
                about: ""
            }
        ]
    });

    useEffect(() => {
        fetch('data/players.json')
            .then(response => response.json())
            .then(json => setState({players: json}));
    });

    return (
        <Container>
            <Row className="justify-content-center">
                <Col lg={8} md={10}>
                    <h1>Players</h1>
                </Col>
            </Row>
            {state.players.map((player, i) => {
                return (
                    <Player key={i} image={player.image} thumbnail={player.thumbnail} alt={player.alt}
                            name={player.name} office={player.office} hometown={player.hometown}
                            nickname={player.nickname} about={player.about}/>
                );
            })}
        </Container>
    );

}
