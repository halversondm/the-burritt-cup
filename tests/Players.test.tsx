import Players from '../src/Players';
import {render} from "@testing-library/react";

test("renders without crashing", () => {
  const unit = <Players />;
  const component = render(unit);
  const tree = component.container;
  expect(tree.toString()).toMatchSnapshot();
});
