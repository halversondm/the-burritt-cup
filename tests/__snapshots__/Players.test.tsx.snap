// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `"<div><div class="container"><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><h1>Players</h1></div></div><div><hr><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><div class="post-preview"><h2 class="post-title"><a href=""><img class="head" alt=""></a> </h2><h4 class="post-subtitle">Hometown:  <br> Nickname:  <br></h4><p class="post-meta"></p></div></div></div></div></div></div>"`;
